packages/block-library/src/comment-template/index.php

```diff
@@ -30,8 +30,20 @@ function block_core_comment_template_render_comments( $comments, $block ) {
 			$context['commentId'] = $comment_id;
 			return $context;
 		};
+
+		/*
+		 * We set commentId context through the `render_block_context` filter so
+		 * that dynamically inserted blocks (at `render_block` filter stage)
+		 * will also receive that context.
+		 */
 		add_filter( 'render_block_context', $filter_block_context );
-		$block_content = $block->render( array( 'dynamic' => false ) );
+
+		/*
+		 * We construct a new WP_Block instance from the parsed block so that
+		 * it'll receive any changes made by the `render_block_data` filter.
+		 */
+		$block_content = ( new WP_Block( $block->parsed_block ) )->render( array( 'dynamic' => false ) );
```

Comment:
I believe that we still need to propagate the available context here to avoid issues like the one reported in https://core.trac.wordpress.org/ticket/58278 by @jdm-web. See also his comment in https://github.com/WordPress/gutenberg/pull/50313#issuecomment-1547741568.

I checked how we instantiate `WP_Block` in WordPress core and it both places the upper context gets passed down:

https://github.com/WordPress/wordpress-develop/blob/d46c47f0fdd43ad67743fb390187c4e32876d8a3/src/wp-includes/blocks.php#L1072

https://github.com/WordPress/wordpress-develop/blob/d46c47f0fdd43ad67743fb390187c4e32876d8a3/src/wp-includes/class-wp-block-list.php#L96